clang/lib/Sema/SemaDecl.cpp

```diff
@@ -1523,7 +1523,10 @@ void Sema::PushOnScopeChains(NamedDecl *D, Scope *S, bool AddToContext) {
 
   // Out-of-line definitions shouldn't be pushed into scope in C++, unless they
   // are function-local declarations.
-  if (getLangOpts().CPlusPlus && D->isOutOfLine() && !S->getFnParent())
+  bool OutOfLine = D->isOutOfLine();
+  if (const auto *ECD = dyn_cast<EnumConstantDecl>(D))
+    OutOfLine = OutOfLine || cast<Decl>(ECD->getDeclContext())->isOutOfLine();
```

Comment:
Hmm... What is the logic to set the `EnumConstantDecl` (or, that is, just the `Decls` themselves) to be `OutOfLine`? I can't imagine this is the only logic for `OutOfLine` that could be reasonable to check, and wonder if we should just set `EnumConstantDecl::OutOfLine` 'correctly'.

WDYT?

Reply:
Hmmm I see that `isOutOfLine()` is actually a virtual function, so yeah, I think I can hide this logic in there for `EnumConstantDecl` itself. Good call!